File: ProgrammingFundamentalsExercises/FilesAndDirectoriesExercises/06.FixEmails/FixEmails.cs
using System;
using System.Collections.Generic;
using System.Linq;

class FixEmails
{
    static void Main(string[] args)
    {
        string command = Console.ReadLine();
        Dictionary<string, string> personEmail = new Dictionary<string, string>();

        while (command != "stop")
        {
            string person = command;
            string email = Console.ReadLine();
            string domain = email.Split('.').Last().ToLower();

            if (domain != "us" && domain != "uk")
            {
                personEmail[person] = email;
            }

            command = Console.ReadLine();
        }

        foreach (KeyValuePair<string, string> pair in personEmail)
        {
            Console.WriteLine($"{pair.Key} -> {pair.Value}");
        }
    }
}

File: JavaAdvanced/SetsAndMapsExercises/src/FixEmails.java
import java.util.LinkedHashMap;
import java.util.Scanner;

public class FixEmails {
    public static void main(String[] args) {
        Scanner console = new Scanner(System.in);

        String command = console.nextLine();

        LinkedHashMap<String, String> personEmail = new LinkedHashMap<>();

        while (!command.equals("stop")) {
            String person = command;
            String email = console.nextLine();

            String[] domainParts = email.split("@")[1].split("\\.");
            String domain = domainParts[domainParts.length - 1].toLowerCase();

            if (!domain.equals("us") && !domain.equals("uk") && !domain.equals("com")) {
                personEmail.put(person, email);
            }

            command = console.nextLine();
        }

        for (String person : personEmail.keySet()) {
            System.out.printf("%s -> %s%n", person, personEmail.get(person));
        }
    }
}